CHANGELOG.md

```diff
@@ -5,6 +5,54 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Register this Doorkeeper extension as `device_code` custom OAuth Grant Flow.
+
+### Changed
+- Upgrade `doorkeeper` dependency, matching versions `~> 5.5`.
+- Use the standard IANA URN value `urn:ietf:params:oauth:grant-type:device_code`
+  as grant type for device access token requests. It replaces the previous
+  value `device_code`, which was deliberately nonstandard to make it work with
+  Doorkeeper versions prior to `5.5`.
+
+  The old `device_code` value might still work out of the box with some
+  Doorkeeper releases, thanks to a fallback strategy; however, a warning message
+  will be printed, and this functionality will be eventually removed in newer
+  Doorkeeper versions.
+
+  An appropriate way to make the device flow work with any nonstandard / custom
+  grant type is to simply register your own custom flow, using the strategy
+  class `Doorkeeper::Request::DeviceCode` provided by this extension, and enable
+  it in Doorkeeper configuration.
+  
+  For example, you could add the following code to an appropriate place, such
+  as an initializer file:
```

Comment:
To be clear: These steps are required to use the gem, right?

The current descriptions sounds like this is an optional setup step for users who want to customize things.

Reply:
The gem works without additional registration needed. The whole point that made me write this, is that it now matches the correct IANA URN grant type (`"urn:ietf:params:oauth:grant-type:device_code"`). So, this new version is not directly back-compatible with clients that are using the old nonstandard grant type (which was just `"device_code"`).

Actually, the currently latest version of Doorkeeper still has the aforementioned fallback strategy in place: using the old grant type still works, but you get deprecation warnings. So the whole point here was to show how you can use a nonstandard grant type (in particular, the "legacy" one from this very gem), either in addition to the new correct one or in place of it.

I was trying to be not too verbose with this, but probably it ended up being not too clear... we can try to improve it, but maybe the whole explanation is just unnecessary.